hal/src/nRF52840/i2c_hal.cpp

```diff
@@ -214,7 +237,7 @@ static int twi_init(HAL_I2C_Interface i2c) {
 
 void HAL_I2C_Init(HAL_I2C_Interface i2c, void* reserved) {
     if (m_i2c_map[i2c].enabled) {
-        twi_uinit(i2c);
+        twi_uninit(i2c);
```

Comment:
As discussed, let's not call uninit here to avoid issues for the I2C peripheral shared between system and user code (the one that has PMIC and FuelGauge).

Going further down let's change:

```cpp
    os_thread_scheduling(false, NULL);
    if (m_i2c_map[i2c].mutex == NULL) {
        os_mutex_recursive_create(&m_i2c_map[i2c].mutex);
    } else {
        // Already initialized
        return;
    }
    HAL_I2C_Acquire(i2c, NULL);
    os_thread_scheduling(true, NULL);
```

Reply:
Should we re-enable schedule here?
```
    os_thread_scheduling(false, NULL);
    if (m_i2c_map[i2c].mutex == NULL) {
        os_mutex_recursive_create(&m_i2c_map[i2c].mutex);
    } else {
        // Already initialized
        os_thread_scheduling(true, NULL);
        return;
    }
    HAL_I2C_Acquire(i2c, NULL);
    os_thread_scheduling(true, NULL);
```

Reply:
Should we re-enable schedule here?
```   
os_thread_scheduling(false, NULL);
    if (m_i2c_map[i2c].mutex == NULL) {
        os_mutex_recursive_create(&m_i2c_map[i2c].mutex);
    } else {
        // Already initialized
        os_thread_scheduling(true, NULL);
        return;
    }
    HAL_I2C_Acquire(i2c, NULL);
    os_thread_scheduling(true, NULL);
```

Reply:
Yes, we should :+1: